compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -3108,12 +3115,16 @@ class Typer extends Namer
         val propFail = propagatedFailure(args)
 
         def issueErrors(): Tree = {
+          lazy val paramSyms = tree.symbol.paramSymss.flatten.map(sym => sym.name -> sym).toMap
+          def paramSymWithMethodTree(paramName: TermName) = if tree.symbol.exists then Some((paramSyms(paramName), tree)) else None
```

Comment:
paramSyms could be a non-lazy local val defined inside paramSymWithMethodTree now.